Rover/sailboat.cpp

```diff
@@ -552,3 +585,111 @@ bool Sailboat::motor_assist_low_wind() const
             rover.g2.windvane.wind_speed_enabled() &&
             (rover.g2.windvane.get_true_wind_speed() < sail_windspeed_min));
 }
+
+bool Sailboat::in_irons()
+{
+   // only enabled for separate mainsail and options bit set
+   if (! get_out_of_in_irons_in_auto_enabled()){
+       clear_in_irons();
+       return false;
+   }
+   // helper lambda functions
+
+   // do low level set in irons work
+   auto set_in_irons = [this]()
```

Comment:
Typically we don't use lambda functions, but we have them in a few places where there is a clear benefit. Only used once? This is only used once?